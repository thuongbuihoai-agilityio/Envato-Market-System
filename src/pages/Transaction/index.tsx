import { memo, lazy } from 'react';
import isEqual from 'react-fast-compare';
import { Box, Flex, Grid, GridItem } from '@chakra-ui/react';

// HOCs
import { withErrorBoundary } from '@app/hocs/withErrorBoundary';
import { Lazy } from '@app/components';

// lazy loading components
const TransactionTable = lazy(() => import('@app/components/TransactionTable'));
const CardPayment = lazy(() => import('@app/components/CardPayment'));
const BoxChat = lazy(() => import('@app/components/BoxChat'));

const Transaction = () => (
  <Grid
    bg="background.body.primary"
    py={12}
    px={{ base: 6, xl: 12 }}
    templateColumns={{ base: 'repeat(1, 1fr)', '2xl': 'repeat(4, 1fr)' }}
    display={{ sm: 'block', md: 'grid' }}
    gap={{ base: 0, '2xl': 12 }}
  >
    <GridItem colSpan={3}>
      <Box
        as="section"
        bgColor="background.component.primary"
        borderRadius={8}
        px={6}
        py={5}
      >
        <TransactionTable />
      </Box>
    </GridItem>
    <GridItem mt={{ base: 6, '2xl': 0 }}>
      <Flex direction={{ base: 'column', lg: 'row', xl: 'column' }} gap={6}>
        <Lazy>
          <CardPayment />
          <BoxChat />
        </Lazy>
      </Flex>
    </GridItem>
  </Grid>
);

const TransactionPage = memo(withErrorBoundary(Transaction), isEqual);

export default TransactionPage;
