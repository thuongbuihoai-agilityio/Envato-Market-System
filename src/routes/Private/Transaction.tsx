import { RouteObject } from 'react-router-dom';

// Constants
import { ROUTES } from '@constants/routers';

// Pages
import Transaction from '@pages/Transaction';

export const transactionRoutes: RouteObject = {
  children: [
    {
      path: ROUTES.TRANSACTION,
      Component: Transaction,
    },
  ],
};
