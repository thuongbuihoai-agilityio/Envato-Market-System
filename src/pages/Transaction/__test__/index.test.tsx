import { render } from '@testing-library/react';

// Pages
import Transaction from '..';

describe('Transaction page', () => {
  it('Match to snapshot', () => {
    const { container } = render(<Transaction />);

    expect(container).toMatchSnapshot();
  });
});
