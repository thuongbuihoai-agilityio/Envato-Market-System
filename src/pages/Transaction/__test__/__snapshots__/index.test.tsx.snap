// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transaction page Match to snapshot 1`] = `<div />`;
